src/http/api/account.ts

```diff
@@ -304,8 +305,34 @@ export function createGetAccountPostsHandler(
                 if (postResult instanceof Error) {
                     throw postResult;
                 }
-                result.results = postResult.results;
-                result.nextCursor = postResult.nextCursor;
+                if (isError(postResult)) {
+                    const error = getError(postResult);
+                    switch (error) {
+                        case 'invalid-handle':
+                            logger.error(`Invalid handle ${handle}`);
+                            return new Response(null, { status: 400 });
+                        case 'missing-default-account':
+                            logger.error('Missing default account');
+                            return new Response(null, { status: 400 });
+                        case 'invalid-next-parameter':
+                            logger.error('Invalid next parameter');
+                            return new Response(null, { status: 400 });
+                        case 'actor-not-found':
+                            logger.error(`Actor not found for ${handle}`);
+                            return new Response(null, { status: 400 });
+                        case 'error-getting-outbox':
+                            logger.error(`Error getting outbox for ${handle}`);
+                            return new Response(null, { status: 500 });
+                        case 'no-page-found':
+                            logger.error(
+                                `No page found in outbox for ${handle}`,
+                            );
+                            return new Response(null, { status: 500 });
+                    }
```

Comment:
Can we add an "exhaustive check" here?

```ts
default:
    const exhaustiveCheck: never = error;
    throw new Error(`Unhandled error ${exhaustiveCheck}`);
```

This ensures that all error types are handled if we ever add new ones! I'll look at getting a helper added to make this simpler